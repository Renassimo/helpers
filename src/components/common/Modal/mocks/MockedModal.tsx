import { FormEvent, ReactNode } from 'react';
import { Breakpoint } from '@mui/material';

const MockedModal = jest.fn(
  ({
    open,
    children,
    onClose,
    onSubmit,
    title,
    loading,
    disabled,
    maxWidth = 'sm',
  }: {
    open: boolean;
    children: ReactNode;
    onClose: () => void;
    onSubmit: (event: FormEvent<HTMLFormElement>) => void;
    title: string;
    loading: boolean;
    disabled?: boolean;
    maxWidth?: Breakpoint;
  }) => (
    <div>
      {children}
      <div>
        title: {title}; maxWidth: {maxWidth};{open ? 'open' : 'closed'};
        {loading ? 'loading' : 'not loading'};
        {disabled ? 'disabled' : 'not disabled'}; onClose{' '}
        {!!onClose ? 'passed' : 'did not passed'}; onSubmit{' '}
        {!!onSubmit ? 'passed' : 'did not passed'};
      </div>
    </div>
  )
);

export default MockedModal;
