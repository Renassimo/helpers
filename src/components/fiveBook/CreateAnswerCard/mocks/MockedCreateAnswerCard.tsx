const MockedCreateAnswerCard = jest.fn(({ children }) => (
  <div>
    Mocked Create Answer Card:
    {children}
  </div>
));

export default MockedCreateAnswerCard;
