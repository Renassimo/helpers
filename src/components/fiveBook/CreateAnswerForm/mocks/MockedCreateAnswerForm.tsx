const MockedCreateAnswerForm = jest.fn(() => (
  <div>Mocked Create Answer Form</div>
));

export default MockedCreateAnswerForm;
