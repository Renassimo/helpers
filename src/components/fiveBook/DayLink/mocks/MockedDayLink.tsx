const MockedDayLink = jest.fn(({ prev, next }) => (
  <div>
    Mocked Day Link: Prev: {prev}
    Next: {next}
  </div>
));

export default MockedDayLink;
