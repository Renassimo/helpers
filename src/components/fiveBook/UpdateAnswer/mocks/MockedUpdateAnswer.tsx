const MockedUpdateAnswer = jest.fn(() => <div>Mocked Update Answer:</div>);

export default MockedUpdateAnswer;
