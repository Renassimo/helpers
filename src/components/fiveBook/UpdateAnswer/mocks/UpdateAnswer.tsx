const MockedUpdateAnswer = jest.fn(() => <div>Mocked Change Answer:</div>);

export default MockedUpdateAnswer;
