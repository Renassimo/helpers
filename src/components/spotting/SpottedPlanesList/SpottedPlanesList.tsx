import React from 'react';

import useSpottedPlanes from '@/hooks/spotting/useSpottedPlanes';

import PlanesContainer from '@/components/spotting/PlanesContainer';
import SpottedPlaneCard from '@/components/spotting/SpottedPlaneCard';
import GroupPlanesModal from '@/components/spotting/GroupPlanesModal';

const SpottedPlanesList = () => {
  const { spottedPlanes } = useSpottedPlanes();

  return (
    <PlanesContainer>
      {spottedPlanes.map((item) => (
        <SpottedPlaneCard data={item} key={item.id} selectable />
      ))}
      <GroupPlanesModal />
    </PlanesContainer>
  );
};

export default SpottedPlanesList;
