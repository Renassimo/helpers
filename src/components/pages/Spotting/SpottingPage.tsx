import { useEffect } from 'react';

import { PageInfo, User } from '@/types/auth';
import { NotionError } from '@/types/notion';
import { SpottedPlaneApiData } from '@/types/spotting';

import SpottingProvider from '@/providers/spotting/SpottingProvider';

import useAlerts from '@/hooks/alerts';

import PageTemplate from '@/components/templates/PageTemplate';
import SpottedPlanesList from '@/components/spotting/SpottedPlanesList';

const SpottingPage = ({
  user,
  pages,
  data,
  error,
}: {
  user: User;
  pages: PageInfo[];
  data: SpottedPlaneApiData[] | null;
  error: NotionError | null;
}) => {
  const { createErrorAlert } = useAlerts();

  useEffect(() => {
    if (error) createErrorAlert(error.message || error.code || error.status);
  }, [createErrorAlert, error]);

  return (
    <SpottingProvider data={data}>
      <PageTemplate title="5book" user={user} pages={pages}>
        <SpottedPlanesList />
      </PageTemplate>
    </SpottingProvider>
  );
};

export default SpottingPage;
