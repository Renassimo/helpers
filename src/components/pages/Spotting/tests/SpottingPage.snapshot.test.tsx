import renderWithTheme from '@/tests/helpers';

import SpottingPage from '@/components/pages/Spotting';
import SpottedPlanesList from '@/spotting/components/SpottedPlanesList';

import MockedSpottedPlanesList from '@/spotting/components/SpottedPlanesList/mocks';

import { mockedSpottedPlaneApiDataTruthy } from '@/types/spotting/mocks';
import { mockedPageInfos, mockedUser } from '@/types/auth/mocks';
import { mockedNotionError418 } from '@/types/notion/mocks';

jest.mock('@/spotting/components/SpottedPlanesList');

describe('SpottingPage', () => {
  beforeEach(() => {
    (SpottedPlanesList as unknown as jest.Mock).mockImplementationOnce(
      MockedSpottedPlanesList
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('when got data', () => {
    test('renders successfully', () => {
      // Arrange
      const mockedData = [mockedSpottedPlaneApiDataTruthy];
      // Act
      const { container } = renderWithTheme(
        <SpottingPage
          user={mockedUser}
          pages={mockedPageInfos}
          data={mockedData}
          error={null}
        />
      );
      // Assert
      expect(container).toMatchSnapshot();
      expect(MockedSpottedPlanesList).toHaveBeenCalledWith({}, {});
    });
  });

  describe('when got error', () => {
    test('renders successfully', () => {
      // Arrange
      // Act
      const { container } = renderWithTheme(
        <SpottingPage
          user={mockedUser}
          pages={mockedPageInfos}
          data={null}
          error={mockedNotionError418}
        />
      );
      // Assert
      expect(container).toMatchSnapshot();
      expect(MockedSpottedPlanesList).toHaveBeenCalledWith({}, {});
    });
  });
});
