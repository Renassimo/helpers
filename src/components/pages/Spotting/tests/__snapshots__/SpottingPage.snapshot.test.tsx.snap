// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpottingPage when got data renders successfully 1`] = `
<div>
  <main
    class="css-3wo33r"
  >
    <div
      class="css-5aiikd"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-s31e50-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-9kk78p-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <a
              href="/"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    alt="Main page"
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/favicon.ico"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <button
              aria-label="Open settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  alt="Jesse Pinkman"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://pic.com"
                />
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div
      class="css-16cpvj6"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
      >
        <div>
          Mocked Spotted Planes List:
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`SpottingPage when got error renders successfully 1`] = `
<div>
  <main
    class="css-3wo33r"
  >
    <div
      class="css-5aiikd"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-s31e50-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-9kk78p-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <a
              href="/"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    alt="Main page"
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/favicon.ico"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <button
              aria-label="Open settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  alt="Jesse Pinkman"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://pic.com"
                />
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div
      class="css-16cpvj6"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
      >
        <div>
          Mocked Spotted Planes List:
        </div>
      </div>
    </div>
  </main>
</div>
`;
